src/Servers/Kestrel/Core/src/Internal/Http2/HPackHeaderWriter.cs

```diff
@@ -99,6 +99,9 @@ private static bool EncodeHeadersCore(DynamicHPackEncoder hpackEncoder, Http2Hea
                 var staticTableId = headersEnumerator.HPackStaticTableId;
                 var name = headersEnumerator.Current.Key;
                 var value = headersEnumerator.Current.Value;
+                var valueEncoding =
+                    ReferenceEquals(headersEnumerator.EncodingSelector, KestrelServerOptions.DefaultHeaderEncodingSelector)
+                    ? null : headersEnumerator.EncodingSelector(name);
```

Comment:
Couldn't this check happen once when the header enumerator is initialized? i.e. only set an encoding selector if it isn't equal to DefaultHeaderEncodingSelector

Reply:
Although... the performance difference between `ReferenceEquals` and `!= null` is probably zero 😬

Reply:
Yeah I don't think it would be a performance difference, at best it would make the code more concise and at worst it might make it easier to miss a code path where we forget to initialize it. I'll leave it for now since it's consistent with how we do it for request headers.

Reply:
I was thinking about this same thing, but I was in no position to talk since I also did it this way for request headers 😅